os/drivers/lcd/mipi_lcd.c

```diff
@@ -236,6 +241,10 @@ static int lcd_putarea(FAR struct lcd_dev_s *dev, fb_coord_t row_start, fb_coord
 	priv->config->lcd_put_area((u8 *)lcd_buffer[lcd_buffer_index], row_start, col_start, row_end, col_end);
 	lcd_buffer_index = (1 - lcd_buffer_index);
 #else
+	if (lcd_data_logo != NULL) {
+		kmm_free(lcd_data_logo);
+		lcd_data_logo = NULL;
+	}
```

Comment:
Could you explain why this is added in this case only?

Reply:
When ROTATION is enabled, then LCD driver already have buffer allocated for rotation purposes.
Since, in initial bootup (when displaying logo in screen), there is no put area call from application. So, we can safely use this rotation buffer to display our logo.
Now, when ROTATION is disabled, then we will need extra buffer which will be used for this specific logo display purpose. 
When application calls the putarea, there is no need of logo now, so this allocated buffer should be deleted.

Reply:
1. Two buffers are necessary in ROTATION only? What about flushing thread? I mean when we need the flushing thread only without rotation, no need two buffers? Maybe not.
2. Could you leave some comment for that? It's difficult to know why it needs.